scripts/quests/sandoria/A_Timely_Visit.lua

```diff
@@ -0,0 +1,202 @@
+-----------------------------------
+-- A Timely Visit
+-----------------------------------
+-- Log ID: 0, Quest ID: 100
+-- Deraquien !pos -98 -2 32
+-- Narvecaint !pos -261 23 127
+--  Phillone !pos -208 -2 67
+--       ??? !pos -310 0 407
+-----------------------------------
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/titles')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+ID = require('scripts/zones/Jugner_Forest/IDs')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.A_TIMELY_VISIT)
+
+quest.reward =
+{
+    item = xi.items.MEDIEVAL_COLLAR,
+    fame = 60,
+    fameArea = xi.quest.fame_area.SANDORIA,
+    title = xi.title.OBSIDIAN_STORM,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getFameLevel(xi.quest.fame_area.SANDORIA) >= 3
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Deraquien'] = quest:progressEvent(33),
+
+            onEventFinish =
+            {
+                [33] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Deraquien'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(80)
```

Comment:
i see these are all set as progress event, when some should be quest:event()
you would only use progressevent if its a. progressing the quest or b. its the only dialog it will ever do on this leg of the quest. otherwise its breaking the reason for the new quest framework with allowing progression of all quests from an npc becuase you are locking out other possible dialogs